src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/Utf8StringMarshaller.cs

```diff
@@ -101,5 +103,26 @@ public void FreeNative()
             if (_allocated != null)
                 Marshal.FreeCoTaskMem((IntPtr)_allocated);
         }
+
+        internal static string? ConvertToManaged(byte* value)
```

Comment:
I am wondering whether this kind of state-less marshalling is common enough to promote it to public pattern. Producing marshaller instances for it comes with a lot of ceremony.

Reply:
Yeah, I agree that the JIT is doing good job eliminating the extra copies for the simple cases where the overhead is noticeable. And the stateless approach is not a good idea for marshallers that hold unmanaged resources since it complicates cleanup. 

There may be still something to be gained marshallers that do not hold any resources and that just do a trivial conversion of the value (e.g. the marshaller for `HRESULT`s). The amount of ceremony for these cases feels too high. We can explore that separately. It is not directly related to this PR.